launchDarkly/client/LDProvider.tsx

```diff
@@ -0,0 +1,26 @@
+'use client';
+
+import { LDOptions } from 'launchdarkly-js-client-sdk';
+import { LDContext } from 'launchdarkly-js-sdk-common';
+import { asyncWithLDProvider } from 'launchdarkly-react-client-sdk';
+import { ReactNode, use } from 'react';
+
+export function LDProvider({
+  children,
+  context,
+  options,
+}: {
+  children: ReactNode;
+  context: LDContext;
+  options: LDOptions;
+}) {
+  const Provider = use(
```

Comment:
This is so cool!

Reply:
Yessss - slight tweak here on this, in the recent version of Next and NextJS 15 - use will throw an error in console (but still work fine). We want to do something like below. This moves the promise out and into the proper place. I'll push an update. 

```javascript 
'use client';

import { LDOptions } from 'launchdarkly-js-client-sdk';
import { LDContext } from 'launchdarkly-js-sdk-common';
import { asyncWithLDProvider } from 'launchdarkly-react-client-sdk';
import { ReactNode, use, useMemo } from 'react';

export function LDProvider({
  children,
  context,
  options,
}: {
  children: ReactNode;
  context: LDContext;
  options: LDOptions;
}) {
  const Provider = useMemo(
    () =>
      asyncWithLDProvider({
        clientSideID: process.env.NEXT_PUBLIC_LD_CLIENT_SDK_KEY ?? '',
        context,
        options,
      }),
    [context, options],
  );

  const LDClientProvider = use(Provider);

  return <LDClientProvider>{children}</LDClientProvider>;
}
```